File: hs/Mod_toBinary.hs
module Mod_toBinary ( toBinary ) where
import Prelude hiding (and, or) 
import Debug.Trace
import Splib
import Types

toBinary :: Int -> [Int]
toBinary from | from == 0 = [0]
              | otherwise = toBinaryHelper from []

toBinaryHelper from res 
   | from == 0 = res
   | otherwise = toBinaryHelper (sp_div from 2) (prepend res (sp_mod from 2))

File: start/cpp/Mod_toBinary.cpp
#pragma once

List<Int> toBinary(int from){
	List<Int> res;
	while(from > 0){
		res = prepend(res, sp_mod(from, 2));
		from = sp_div(from, 2);
	}
	if(empty(res))
		res = {0};
	return res;
}